TruePath/PathStrings.cs

```diff
@@ -36,61 +38,100 @@ public static class PathStrings
     ///     Note that this operation will never perform any file IO, and is purely string manipulation.
     /// </para>
     /// </summary>
+    [SkipLocalsInit] // is necessary to prevent the CLR from filling stackalloc with zeros.
     public static string Normalize(string path)
     {
-        // TODO[#19]: Optimize this. It is possible to do with less allocations.
-        var segments = new List<(int Start, int End)>();
+        int written = 0;
 
-        int? currentSegmentStart = 0;
-        for (var i = 0; i < path.Length; i++)
+        char[]? array = path.Length < (IntPtr.Size == 4 ? 512 : 4096) ? null : ArrayPool<char>.Shared.Rent(path.Length);
+
+        Span<char> normalized = array != null ? array.AsSpan() : stackalloc char[path.Length];
```

Comment:
```suggestion
        char[]? array = null;
        var buffer = path.Length <= 512
            ? stackalloc char[512]
            : array = ArrayPool<char>.Shared.Rent(path.Length);
```

Reply:
Excessive stack pressure is undesirable as you may not know how deep is the current call stack or how deep it still needs to go. There is no reason to risk SO on systems with low stack limits given beyond 512B-1KB ArrayPool becomes the more profitable option if you assume you are not using unsafe with skip locals init.